contracts/examples/mintable-erc1155-transfer/FxMintableERC1155ChildTunnel.sol

```diff
@@ -0,0 +1,242 @@
+        address _childTokenTemplate,
+        address _rootTokenTemplate
+    ) FxBaseChildTunnel(_fxChild) {
+        require(
+            Address.isContract(_childTokenTemplate),
+            "FxMintableERC1155ChildTunnel: Token template is not contract"
+        );
+        childTokenTemplate = _childTokenTemplate;
+        // compute root token template code hash
+        rootTokenTemplateCodeHash = keccak256(minimalProxyCreationCode(_rootTokenTemplate));
+    }
+
+    //
+    // External methods
+    //
+
+    // deploy child token with unique id
+    function deployChildToken(bytes32 _uniqueId, string calldata _uri) external {
+        // deploy new child token using unique id
+        address childToken = createClone(keccak256(abi.encodePacked(_uniqueId)), childTokenTemplate); // child salt, childTokenTemplate
+
+        // compute root token address before deployment using create2
+        address rootToken = computedCreate2Address(
+            keccak256(abi.encodePacked(childToken)), // root salt
+            rootTokenTemplateCodeHash,
+            fxRootTunnel
+        );
+
+        // check if mapping is already there
+        require(rootToChildToken[rootToken] == address(0x0), "FxMintableERC1155ChildTunnel: ALREADY_MAPPED");
+        rootToChildToken[rootToken] = childToken;
+        emit TokenMapped(rootToken, childToken);
+
+        // initialize child token with all parameters
+        FxMintableERC1155(childToken).initialize(address(this), rootToken, _uri, msg.sender);
+    }
+
+    function withdraw(
+        address childToken,
+        uint256 id,
+        uint256 amount,
+        bytes calldata data
+    ) public {
+        _withdraw(childToken, msg.sender, id, amount, data);
+    }
+
+    function withdrawTo(
+        address childToken,
+        address receiver,
+        uint256 id,
+        uint256 amount,
+        bytes calldata data
+    ) public {
+        _withdraw(childToken, receiver, id, amount, data);
+    }
+
+    function withdrawBatch(
+        address childToken,
+        uint256[] calldata ids,
+        uint256[] calldata amounts,
+        bytes calldata data
+    ) public {
+        _withdrawBatch(childToken, msg.sender, ids, amounts, data);
+    }
+
+    function withdrawToBatch(
+        address childToken,
+        address receiver,
+        uint256[] calldata ids,
+        uint256[] calldata amounts,
+        bytes calldata data
+    ) public {
+        _withdrawBatch(childToken, receiver, ids, amounts, data);
+    }
+
+    //
+    // Internal methods
+    //
+
+    function _processMessageFromRoot(
+        uint256, /* stateId */
+        address sender,
+        bytes memory data
+    ) internal override validateSender(sender) {
+        (bytes32 syncType, bytes memory syncData) = abi.decode(data, (bytes32, bytes));
+
+        if (syncType == DEPOSIT) {
+            _syncDeposit(syncData);
+        } else if (syncType == DEPOSIT_BATCH) {
+            _syncDepositBatch(syncData);
+        } else {
```

Comment:
`FxMintableERC721RootTunnel` contract will send `MAP_TOKEN` message but it will end up reverting in child tunnel, hence `childToken` can never be deployed onto the chain.

Reply:
Removed MAP_TOKEN message from root tunnel in bd287b9ea746c1b69d7a08d174a10003d15bc70c [here](https://github.com/fx-portal/contracts/commit/bd287b9ea746c1b69d7a08d174a10003d15bc70c#diff-a9017e0f48ce974c782326093f8ea41b2ab2ff73c4d4108711881d025a9ab817L15) and added deployRootToken on rootTunnel.